stix/migrator.py

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (C) 2022 Vaticle
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import json
+import os
+
+from stix.query import StixInsertGenerator
+from stix.typedb_inserter import TypeDBInserter
+
+
+class StixMigrator:
+
+    def __init__(self, typedb_uri, database, batch_size, num_threads):
+        self.inserter = TypeDBInserter(typedb_uri, database, batch_size=batch_size, num_threads=num_threads)
+
+    def migrate(self, data_path):
+        print('.....')
+        print('Inserting data...')
+        print('.....')
+        stix_json_objects = self._read_stix_objects_json(data_path)
+        insert_generator = StixInsertGenerator(stix_json_objects)
+        self._migrate_stix_objects(insert_generator)
+        self._migrate_stix_relationships(insert_generator)
+        self._migrate_kill_chain_phases(insert_generator)
+        self._migrate_external_references(insert_generator)
+        print('.....')
+        print('Successfully inserted data!')
+        print('.....')
```

Comment:
this is the real "work" entry point. We could run this on multiple `data_path` and reuse this object!